recipes/cpppo/meta.yaml

```
@@ -0,0 +1,54 @@
+{% set name = "cpppo" %}
+{% set version = "4.4.2" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/cpppo-{{ version }}.tar.gz
+  sha256: 8c5a2947d6e9fd732d2d887fa02216beebb4068ab99e2590590f491425992bc6
+  patches:
+    - remove_getattr_entrypoint.diff
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  entry_points:
+    - enip_server = cpppo.server.enip.main:main
+    - enip_client = cpppo.server.enip.client:main
+    - enip_get_attribute = cpppo.server.enip.get_attribute:main
+
+requirements:
+  host:
+    - python >=3.6
+    - pip
+  run:
+    - python >=3.6
```

```suggestion
    - python >=3.0
    - pip
  run:
    - python >=3.0
```
Technically, it says it's compatible with python>=2.7, but that requires a backport.

---

I don't mind using 3.0 but I thought conda-forge encouraged to pin to the "supported" version of Python. Isn't that the case?

---

I'm not familiar with the reasons behind that policy.

Since this package is noarch, I don't see why providing the version spec stated by upstream is a problem. If this wasn't noarch, then I would understand that reasoning.

---

AFAIK version constraints on noarch: python packages are completely optional. The docs imply that need not constrain python if the package supports all versions.

https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=noarch%20python#noarch-python

---

Thanks for the merge!
The link above above with "no python specific requirement" is a condition for a package to be noarch.

I think the minimum python version for noarch package is to help conda to reduce the search space when resolving environment. But I don't see anything about that in the documentation so it might be something that was removed or that I misunderstood :-)

---

Just my two cents but the 3.6 restriction was indeed for helping the search. Since we now have mamba, I don‘t think this is a problem anymore since it’s generally more performant. 
It never was in the docs though, was just passed around knowledge.😄